# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/back/wgsl/mod.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/naga/src/back/wgsl/mod.rs b/third_party/rust/naga/src/back/wgsl/mod.rs
--- a/third_party/rust/naga/src/back/wgsl/mod.rs
+++ b/third_party/rust/naga/src/back/wgsl/mod.rs
@@ -21,8 +21,23 @@ pub fn write_string(
     module: &crate::Module,
     info: &crate::valid::ModuleInfo,
 ) -> Result<String, Error> {
     let mut w = Writer::new(String::new());
     w.write(module, info)?;
     let output = w.finish();
     Ok(output)
 }
+
+impl crate::AtomicFunction {
+    fn to_wgsl(self) -> &'static str {
+        match self {
+            Self::Add => "Add",
+            Self::And => "And",
+            Self::InclusiveOr => "Or",
+            Self::ExclusiveOr => "Xor",
+            Self::Min => "Min",
+            Self::Max => "Max",
+            Self::Exchange { compare: None } => "Exchange",
+            Self::Exchange { .. } => "CompareExchangeWeak",
+        }
+    }
+}